AdyenCard/Components/Card/CardAnalyticsConfiguration.swift

```diff
@@ -0,0 +1,74 @@
+//
+// Copyright (c) 2024 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+@_spi(AdyenInternal) import Adyen
+
+internal struct CardAnalyticsConfiguration: AnalyticsStringDictionaryConvertible {
+    
+    private enum Constants {
+        static let stringSeparator = ","
+    }
+    
+    private let billingAddressMode: String?
+    private let billingAddressAllowedCountries: String?
+    private let billingAddressRequired: Bool
+    private let showsHolderNameField: Bool
+    private let showsStorePaymentMethodField: Bool
+    private let showsSecurityCodeField: Bool
+    private let showKCPType: String
+    private let socialSecurityNumberMode: String
+    private let enableStoredDetails: Bool
+    private let hasInstallmentOptions: Bool
+    private let allowedCardTypes: String?
+    
+    internal init(configuration: CardComponent.Configuration) {
+        self.billingAddressMode = configuration.billingAddress.mode.analyticsDescription
+        self.billingAddressAllowedCountries = configuration.billingAddress.countryCodes?.joined(separator: Constants.stringSeparator)
+        if case .required = configuration.billingAddress.requirementPolicy {
```

Comment:
Suggestion:

```swift
self.billingAddressRequired = (configuration.billingAddress.requirementPolicy == .required)
```

Reply:
Or maybe even `self.billingAddressRequired = configuration.billingAddress.requirementPolicy.isRequired`
You know I like extensions 😅